File: content/Учёба/Практика/3.2.2026/Практика/PYTHON/5.py
import sympy as sp
import numpy as np
import matplotlib.pyplot as plt

x_sym = sp.Symbol('x')
y_sym = sp.log(sp.cos(x_sym))
dy_dx = sp.diff(y_sym, x_sym)
L_expr = sp.sqrt(1 + dy_dx**2).simplify()
L_sym = sp.integrate(L_expr, (x_sym, 0, sp.pi/4))
L_val = float(L_sym.evalf())
print(f"Аналитический интеграл длины: {L_sym}")
print(f"Численное значение: {L_val:.4f}")
x_num = np.linspace(0, np.pi/4, 100)
y_num = np.log(np.cos(x_num))
plt.figure(figsize=(8, 6))
plt.plot(x_num, y_num, 'b-', linewidth=2, label='y = ln(cos x)')
plt.scatter([0, np.pi/4], [np.log(np.cos(0)), np.log(np.cos(np.pi/4))], color='red', zorder=5, label='Границы')
plt.title(f'Длина дуги. L = {L_val:.4f}')
plt.xlabel('X')
plt.ylabel('Y')
plt.grid(True)
plt.legend()
plt.show()

File: content/Учёба/Практика/3.2.2026/Практика/MATLAB/5.m
syms x_sym;
y_sym = log(cos(x_sym));

% Формула длины дуги: L = int( sqrt(1 + (dy/dx)^2) )

dy_dx = diff(y_sym, x_sym);
L_expr = sqrt(1 + dy_dx^2);

% Символьное интегрирование

L_sym = int(L_expr, x_sym, 0, pi/4);
L_val = double(L_sym);
disp('--- Результат вычисления длины дуги ---');
disp(['Аналитический вид интеграла длины: ', char(L_sym)]);
fprintf('Точное числовое значение: L = %.4f\n', L_val);
x_num = linspace(0, pi/4, 100);
y_num = log(cos(x_num));
figure;
plot(x_num, y_num, 'b-', 'LineWidth', 2);
hold on; % это чтобы было 2 графика на одном холсте

% Отметим границы интегрирования красными узлами

plot(0, log(cos(0)), 'ro', 'MarkerSize', 8, 'MarkerFaceColor', 'r');
plot(pi/4, log(cos(pi/4)), 'ro', 'MarkerSize', 8, 'MarkerFaceColor', 'r');
grid on;
title(['Длина дуги y = ln(cos x) на [0, \pi/4]. L = ', num2str(L_val, '%.3f')]);
xlabel('Ось X');
ylabel('Ось Y');
legend('Кривая', 'Границы интегрирования', 'Location', 'southwest');